src/Katla/HTML.idr

```diff
@@ -99,9 +99,35 @@ standalonePre config = """
     .IdrisLineNumber:target {
       color: gray;
     }
+    .IdrisHighlight {
+      background-color: yellow;
+    }
     </style>
+    <script>
+    function handleHash() {
+      if (!location.hash) return
+      let m = location.hash.match(/#(line\\d+)(?:-(line\\d+))?/)
+      if (m) {
+        let start = document.getElementById(m[1])
+        let end = document.getElementById(m[2])
+        if (start) {
+          start.scrollIntoView()
+          let parent = start.parentElement
+          let lines = parent.children
+          let className = ''
+          for (let n = 0; n < lines.length; n++) {
+            let el = lines[n]
+            if (el === start) className = 'IdrisHighlight'
+            el.className = className
+            if (el === end || className && !end) className = ''
```

Comment:
Can we not immediately `break` here?

Reply:
We are assuming that start <= end and it highlights to the end of the file if that is not the case.  It's pretty easy to fix that without parsing, because we can ask the relative order of the two elements and swap if necessary.

It's not breaking (my first version did) to handle the case where the user clicks on a line number - the URL will change, and we have to clear out the previous highlights and select the newly selected element. The original version, which didn't handle the URL changing, was much simpler.

Also, I forgot to update the test.

Reply:
With a little additional code, we could also support shift-clicking for selecting ranges.  I didn't know if the use case was deep linking into the source, or a user selecting some lines and sharing the link.

Reply:
Ah I see, makes sense re:clearing!

My personal use case is deep linking and I don't mind manually crafting urls
but I could see users quickly wanting the drag-to-select behaviour. :D